packages/plugins/other/visitor-plugin-common/src/selection-set-to-object.ts

```diff
@@ -314,29 +339,64 @@ export class SelectionSetToObject<Config extends ParsedDocumentsConfig = ParsedD
     const possibleTypes = getPossibleTypes(this._schema, this._parentSchemaType);
 
     if (!this._config.mergeFragmentTypes || this._config.inlineFragmentTypes === 'mask') {
-      const grouped = possibleTypes.reduce((prev, type) => {
+      const grouped = possibleTypes.reduce<GroupedStringifedTypes>((prev, type) => {
         const typeName = type.name;
         const schemaType = this._schema.getType(typeName);
 
         if (!isObjectType(schemaType)) {
           throw new TypeError(`Invalid state! Schema type ${typeName} is not a valid GraphQL object!`);
         }
 
-        const selectionNodes = selectionNodesByTypeName.get(typeName) || [];
+        const allNodes = selectionNodesByTypeName.get(typeName) || [];
 
         prev[typeName] ||= [];
 
+        const { incrementalNodes, selectionNodes } = allNodes.reduce<{
+          selectionNodes: (SelectionNode | FragmentSpreadUsage)[];
+          incrementalNodes: FragmentSpreadUsage[];
+        }>(
+          (acc, node) => {
+            if ('fragmentDirectives' in node && hasIncrementalDeliveryDirectives(node.fragmentDirectives)) {
+              acc.incrementalNodes.push(node);
+            } else {
+              acc.selectionNodes.push(node);
+            }
+            return acc;
+          },
+          { selectionNodes: [], incrementalNodes: [] }
+        );
```

Comment:
When a fragment is deferred, I'm collecting it to an `incrementalNodes` array to process differently (to generate an union)